gossip/state/state_test.go

```diff
@@ -786,13 +779,22 @@ func TestHaltChainProcessing(t *testing.T) {
 	v.On("Validate").Return(&errors2.VSCCExecutionFailureError{
 		Err: errors.New("foobar"),
 	}).Once()
-	peerNode := newPeerNodeWithGossipWithValidator(0, mc, noopPeerIdentityAcceptor, g, v)
+
+	buf := bytes.Buffer{}
+
+	logger := flogging.MustGetLogger(gutil.StateLogger).WithOptions(zap.Hooks(func(entry zapcore.Entry) error {
+		buf.WriteString(entry.Message)
+		buf.WriteString("\n")
+		return nil
+	}))
+	peerNode := newPeerNodeWithGossipWithValidator(logger, 0, mc, noopPeerIdentityAcceptor, g, v)
 	defer peerNode.shutdown()
 	gossipMsgs <- newBlockMsg(1)
 
-	assertLogged(t, buf, "Got error while committing")
-	assertLogged(t, buf, "Aborting chain processing")
-	assertLogged(t, buf, "foobar")
+	gom := gomega.NewGomegaWithT(t)
+	gom.Eventually(buf.String, time.Minute).Should(gomega.ContainSubstring("Got error while committing"))
+	gom.Eventually(buf.String, time.Minute).Should(gomega.ContainSubstring("Aborting chain processing"))
+	gom.Eventually(buf.String, time.Minute).Should(gomega.ContainSubstring("foobar"))
```

Comment:
This is not thread safe relative to writing. Growing the buffer will race with String().

Since you're pulling in gomega, you might as well use the gbytes.Buffer you had and the Say matcher.

Reply:
oops, nice catch. Fixed.